swap/swap.go

```diff
@@ -253,32 +253,30 @@ func (s *Swap) handleEmitChequeMsg(ctx context.Context, p *Peer, msg *EmitCheque
 	}
 
 	// submit cheque and cash in async, otherwise this blocks here until the TX is mined
-	go defaultSubmitChequeAndCash(s, otherSwap, opts, actualAmount, cheque)
+	go defaultCashCheque(s, otherSwap, opts, cheque)
 
 	return err
 }
 
-// submitChequeAndCash should be called async as it blocks until the transaction(s) are mined
-// The function submits the cheque to the blockchain and then cashes it in directly
-func submitChequeAndCash(s *Swap, otherSwap contract.Contract, opts *bind.TransactOpts, actualAmount uint64, cheque *Cheque) {
+// cashCheque should be called async as it blocks until the transaction(s) are mined
+// The function cashes the cheque by sending it to the blockchain
+func cashCheque(s *Swap, otherSwap contract.Contract, opts *bind.TransactOpts, cheque *Cheque) {
 	// blocks here, as we are waiting for the transaction to be mined
-	receipt, err := otherSwap.SubmitChequeBeneficiary(opts, s.backend, big.NewInt(int64(cheque.Serial)), big.NewInt(int64(cheque.Amount)), big.NewInt(int64(cheque.Timeout)), cheque.Signature)
+	result, receipt, err := otherSwap.CashChequeBeneficiary(opts, s.backend, s.owner.Contract, big.NewInt(int64(cheque.CumulativePayout)), cheque.Signature)
 	if err != nil {
 		// TODO: do something with the error
 		// and we actually need to log this error as we are in an async routine; nobody is handling this error for now
-		log.Error("error submitting cheque", "err", err)
+		log.Error("error cashing cheque", "err", err)
 		return
 	}
-	log.Debug("submit tx mined", "receipt", receipt)
 
-	receipt, err = otherSwap.CashChequeBeneficiary(opts, s.backend, s.owner.Contract, big.NewInt(int64(actualAmount)))
-	if err != nil {
-		// TODO: do something with the error
-		// and we actually need to log this error as we are in an async routine; nobody is handling this error for now
-		log.Error("error cashing cheque", "err", err)
+	if result.Bounced {
+		log.Error("cheque bounced", "tx", receipt.TxHash)
 		return
+		// TODO: do something here
```

Comment:
Yes. I originally planned to address some of the error handling already in this PR but I thought it would be better to merge this quickly so that other PRs don't have to build on top of the old `submit` / `serial` code in the meantime.